community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/RecoveryIndexingUpdatesValidator.java

```diff
@@ -62,12 +64,18 @@ public RecoveryIndexingUpdatesValidator( PrimitiveLongVisitor<RuntimeException>
     public ValidatedIndexUpdates validate( TransactionRepresentation transaction ) throws IOException
     {
         // Extract updates...
-        extractor.begin( new TransactionToApply( transaction ), new LockGroup() );
-        transaction.accept( extractor );
+        try ( TransactionApplier txApplier = extractor.startTx( new TransactionToApply( transaction ) ) )
+        {
+            transaction.accept( txApplier );
 
-        // and add them to the updates already existing in this batch.
-        extractor.visitUpdatedNodeIds( nodeIdCollector );
-        return this;
+            // and add them to the updates already existing in this batch.
+            extractor.visitUpdatedNodeIds( nodeIdCollector );
+            return this;
+        }
+        catch ( Exception e )
+        {
+            throw new IOException( e );
+        }
```

Comment:
I think IOException is too restrictive. The "outer" catch in commit process catches Throwable anyway any should panic on any exception, not just IOException, so I think it's better with Exception here.